neutron/convert.cpp

```diff
@@ -46,85 +31,48 @@ const auto tofToDSpacingPhysicalConstants =
     2.0 * boost::units::si::constants::codata::m_n * m_to_angstrom /
     (boost::units::si::constants::codata::h * tof_to_s);
 
-namespace neutron {
-namespace tof {
-Dataset tofToDSpacing(const Dataset &d) {
-  if (d.contains(Coord::Ei) || d.contains(Coord::Ef))
-    throw std::runtime_error(
-        "Dataset contains Coord::Ei or Coord::Ef. "
-        "However, conversion to Dim::DSpacing is currently "
-        "only supported for elastic scattering.");
-
+Dataset tofToDSpacing(Dataset &&d) {
   // 1. Compute conversion factor
-  const auto &compPos = d.get(Coord::ComponentInfo)[0](Coord::Position);
+  // TODO Is this a good way to store component information?
```

Comment:
Free helper functions are certainly part of the solution. However this also goes a bit beyond this. We have to decide in what way we want to support operations, and also need to think about creation.

For example, by making the positions in the "component_info" dataset "data" (instead of coords) we can add two component-infos, adding the positions... but for others like rotations that does not make sense. So maybe it should be labels instead? This is just one example.